nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1085,6 +1087,28 @@ private static Object toEnum(Object value, EnumDataType dataType, String fieldNa
         throw new IllegalTypeConversionException("Cannot convert value [" + value + "] of type " + value.getClass() + " to Date for field " + fieldName);
     }
 
+    /**
+     * Converts a java.sql.Date object with 00:00:00 in local time zone (typically coming from a java.sql.ResultSet)
+     * to UTC normalized form (storing epoch corresponding to UTC 00:00:00 on the given day).
+     *
+     * @param date java.sql.Date with local time zone 00:00:00
+     * @return java.sql.Date with UTC 00:00:00
+     */
+    public static Date convertDateToUTC(Date date) {
+        return new Date(date.toLocalDate().atStartOfDay(ZoneId.of("UTC")).toInstant().toEpochMilli());
```

Comment:
Can the `ZoneId.of("UTC")` reference can be replaced with `ZoneOffset.UTC`?

Reply:
`atStartOfDay()` has `ZoneId` parameter only.
In my understanding `ZoneOffset` is DST-agnostic, so the `ZoneId` is the right choice in any case.

Reply:
Reading the documentation, `ZoneOffset` extends `ZoneId` so would `ZoneOffset.UTC` function the same as `ZoneId.of("UTC")`?

Reply:
You are right. I did not notice that `ZoneOffset` extends `ZoneId`. Just focused on that `ZoneId` references an offset and the DST info.
As there is no DST in UTC, it is a simple offset so `ZoneOffset.UTC` will be fine. Thanks!